packages/ui/src/components/networks/NetworkDisplay.tsx

```diff
@@ -0,0 +1,168 @@
+        () => ({
+            accept: isTestnet ? "testnet" : "mainnet",
+            hover(item: NetworkCardProps, monitor: any) {
+                if (!ref.current) {
+                    return
+                }
+
+                const draggedIndex = item.index
+                const hoveredOnIndex = index
+
+                if (draggedIndex === hoveredOnIndex) {
+                    return
+                }
+
+                const hoverBoundingRect = ref.current?.getBoundingClientRect()
+                const hoverMiddleY =
+                    (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2
+                const hoverActualY =
+                    monitor.getClientOffset().y - hoverBoundingRect.top
+
+                // if dragging down, continue only when hover is smaller than middle Y
+                if (
+                    draggedIndex < hoveredOnIndex &&
+                    hoverActualY < hoverMiddleY
+                )
+                    return
+                // if dragging up, continue only when hover is bigger than middle Y
+                if (
+                    draggedIndex > hoveredOnIndex &&
+                    hoverActualY > hoverMiddleY
+                )
+                    return
+
+                if (draggedIndex !== hoveredOnIndex) {
+                    moveCard(draggedIndex, hoveredOnIndex, isTestnet)
+                    item.index = hoveredOnIndex
+                }
+            },
+            collect(monitor) {
+                if (monitor.didDrop()) {
+                    const dropResult =
+                        monitor.getDropResult() as NetworkCardProps
+                    if (
+                        dropResult.index === index &&
+                        dropResult.isTestnet === isTestnet
+                    ) {
+                        setDropAnimation(true)
+                    }
+                }
+            },
+            drop(item: NetworkCardProps) {
+                return item
+            },
+        }),
+        [moveCard, index]
+    )
+
+    const [{ isDragging }, drag] = useDrag(
+        () => ({
+            type: isTestnet ? "testnet" : "mainnet",
+            item: { index, isTestnet },
+            collect: (monitor: DragSourceMonitor) => ({
+                isDragging: monitor.isDragging(),
+            }),
+            end: (item: NetworkCardProps, monitor: DragSourceMonitor) => {
+                const didDrop = monitor.didDrop()
+                if (!didDrop) {
+                    moveCard(item.index, index, isTestnet)
+                }
+            },
+        }),
+        [index, moveCard]
+    )
+
+    drag(drop(ref))
+
+    useEffect(() => {
+        if (dropAnimation) {
+            setTimeout(() => {
+                setDropAnimation(false)
+            }, 1500)
+        }
+    }, [dropAnimation])
+
+    return (
+        <div
+            onClick={onClick}
+            className={`hover:bg-gray-100 hover:cursor-pointer rounded-md cursor-pointer ${
```

Comment:
this kind of condition can be simplified using the `classnames` lib. For example:
```jsx
            className={classnames(
                "hover:bg-gray-100 hover:cursor-pointer rounded-md cursor-pointer",
                isDragging && "bg-slate-600",
                dropAnimation && "bg-green-100 transition-colors"
            )}
```